game-core/src/main/java/games/strategy/engine/auto/update/UpdatedMapsCheck.java

```diff
@@ -1,30 +1,53 @@
 package games.strategy.engine.auto.update;
 
-import java.time.LocalDateTime;
-import java.time.temporal.ChronoField;
+import java.time.LocalDate;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
 
-import com.google.common.base.Strings;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Splitter;
 
 import games.strategy.triplea.settings.ClientSetting;
+import games.strategy.triplea.settings.GameSetting;
 
-class UpdatedMapsCheck {
+final class UpdatedMapsCheck {
+  private UpdatedMapsCheck() {}
 
-  static boolean shouldRunMapUpdateCheck() {
+  static boolean isMapUpdateCheckRequired() {
+    return isMapUpdateCheckRequired(
+        LocalDate.now(),
+        ClientSetting.TRIPLEA_LAST_CHECK_FOR_MAP_UPDATES,
+        ClientSetting::flush);
+  }
+
+  @VisibleForTesting
+  static boolean isMapUpdateCheckRequired(
+      final LocalDate now,
+      final GameSetting updateCheckDateSetting,
+      final Runnable flushSetting) {
     // check at most once per month
-    final LocalDateTime locaDateTime = LocalDateTime.now();
-    final int year = locaDateTime.get(ChronoField.YEAR);
-    final int month = locaDateTime.get(ChronoField.MONTH_OF_YEAR);
-    // format year:month
-    final String lastCheckTime = ClientSetting.TRIPLEA_LAST_CHECK_FOR_MAP_UPDATES.value();
-
-    if (!Strings.nullToEmpty(lastCheckTime).trim().isEmpty()) {
-      final String[] yearMonth = lastCheckTime.split(":");
-      if (Integer.parseInt(yearMonth[0]) >= year && Integer.parseInt(yearMonth[1]) >= month) {
+    final String encodedUpdateCheckDate = updateCheckDateSetting.value();
+    if (!encodedUpdateCheckDate.trim().isEmpty()) {
+      final LocalDate updateCheckDate = parseUpdateCheckDate(encodedUpdateCheckDate);
+      if (updateCheckDate.until(now, ChronoUnit.MONTHS) < 1) {
```

Comment:
Agreed that the readability of these expressions can be improved.  Please see cc291a7a8 and let me know if that's what you were thinking.

Otherwise, the only other thing I can think of is to extract the expression to a named method.  For example, for `UpdatedMapsCheck`:

```java
if (isUpdateCheckLessThanOneMonthOld(updateCheckDate, now)) {
```

and for the expression in `EngineVersionCheck`:

```java
if (isUpdateCheckLessThanTwoDaysOld(updateCheckDate, now)) {
```